### DISCUSSION TITLE
In ring, how can I know which node (shard) will a key go?

### DISCUSSION THREAD

**Discussion by juancresc** [redis/go-redis#2282] [Q&A]
As the question, I'd like to know which shard will a particular key will use. Thanks!

**Comment by maqdev** [comment]
1. key corresponds to some hash slot, this is from documentation: `There are 16384 hash slots in Redis Cluster, and to compute the hash slot for a given key, we simply take the CRC16 of the key modulo 16384.` - don't forget about key tags `{` `}`
2. shards (nodes) handle ranges of slots assigned to them and this information can be retrieved using `CLUSTER NODES` command

**Reply by juancresc** [reply]
I wanted to know in real time, ended up forking and getting the correspondent key since iirc it was a private field
